job-dsl-plugin/src/test/groovy/javaposse/jobdsl/plugin/ExecuteDslScriptsSpec.groovy

```diff
@@ -542,7 +557,31 @@ folder('folder-a/folder-b') {
         jenkinsRule.jenkins.getItemByFullName('/folder/test-job') == null
     }
 
-    def 'only use last build to calculate items to be deleted'() {
+    def 'shelve a project instead of deleting it'() {
+        setup:
+        FreeStyleProject job = jenkinsRule.createFreeStyleProject('seed')
+
+        when:
+        String script1 = 'job("test-job")'
+        ExecuteDslScripts builder1 = new ExecuteDslScripts(script1)
+        builder1.removedJobAction = RemovedJobAction.DELETE
+        runBuild(job, builder1)
+
+        then:
+        jenkinsRule.jenkins.getItemByFullName('test-job') instanceof FreeStyleProject
+
+        when:
+        String script2 = 'job("different-job")'
+        ExecuteDslScripts builder2 = new ExecuteDslScripts(script2)
+        builder2.removedJobAction = RemovedJobAction.SHELVE
+        runBuild(job, builder2)
+
+        then:
+        jenkinsRule.jenkins.getItemByFullName('different-job') instanceof FreeStyleProject
+       // jenkinsRule.jenkins.getItemByFullName('test-job') == null
```

Comment:
Since deletion of parent folders doesn't work as long as the folder contains jobs, I had to change this to wait for the shelve task to finish and then delete any parent folders. That way I could actually test that the shelved project is gone afterwards.

I'm not 100% sure if the waiting for the shelve task completion will work in all scenarios. 
`ExecuteDslScripts` is a build step, adding the `ShelveProjectTask` to the build queue and waiting for it to finish. AFAIU, the queue will always run `Queue.FlyweightTask` instances regardless of the number of executors configured. But I'm not a Jenkins developer, so if you have any input on that, please tell.

I don't know how to test that folders are getting deleted since "test-job" and "different-job" both have the same parent folder. One of them always exists, so the folder is never getting deleted.